apps/taxes/src/pages/api/webhooks/order-fulfilled.ts

```diff
@@ -0,0 +1,63 @@
+import { SaleorAsyncWebhook } from "@saleor/app-sdk/handlers/next";
+import {
+  OrderFulfilledEventSubscriptionFragment,
+  UntypedOrderFulfilledSubscriptionDocument,
+} from "../../../../generated/graphql";
+import { saleorApp } from "../../../../saleor-app";
+import { createLogger } from "../../../lib/logger";
+import { getActiveTaxProvider } from "../../../modules/taxes/active-tax-provider";
+export const config = {
+  api: {
+    bodyParser: false,
+  },
+};
+
+type OrderFulfilledPayload = Extract<
+  OrderFulfilledEventSubscriptionFragment,
+  { __typename: "OrderFulfilled" }
+>;
+
+export const orderFulfilledAsyncWebhook = new SaleorAsyncWebhook<OrderFulfilledPayload>({
+  name: "OrderFulfilled",
+  apl: saleorApp.apl,
+  event: "ORDER_FULFILLED",
+  query: UntypedOrderFulfilledSubscriptionDocument,
+  webhookPath: "/api/webhooks/order-fulfilled",
+});
+
+export default orderFulfilledAsyncWebhook.createHandler(async (req, res, ctx) => {
+  const logger = createLogger({ event: ctx.event });
+  const { payload } = ctx;
+  logger.info({ payload }, "Handler called with payload");
+
+  try {
+    const appMetadata = payload.recipient?.privateMetadata ?? [];
```

Comment:
if metadata is not set ([]) does it make sense to continue code execution?

Reply:
Good point

Reply:
I added validation for that in `getActiveTaxProvider`